slackapi/bolt-python
# Add basic slack CLI support

The goal of this PR is to introduce **basic** support for slack CLI in Bolt for python. This is accomplished by implementing the `get-hooks`, `get-manifest` and `start` interface. Support for `default` and `message-boundaries` communication protocol was also implemented in order to leverage the full capabilities of the slack CLI

#### Testing

1. `slack create -t https://github.com/slack-samples/bolt-python-starter-template -b cli-support`
2. (Optional) In your project set up a venv with python -m venv .venv and source .venv/bin/activate
3. `pip install -r requirements.txt`
4. `pip uninstall slack-bolt`
5. Pull the latest PR branch of bolt python to your local
6. In the Bolt-Python project run `scripts/build_pypi_package.sh` this will generate a `.whl` in the `dist/` folder
8. Back in your project run `pip install global/path/to/bolt-python/dist/something.whl`
9. Add the following fields in the `manifest.json`
    - `"outgoing_domains": []`
    - `"settings"."function_runtime": "remote"`
    - `"settings"."org_deploy_enabled": true`
10. Now you should be able to run the following 
    - `slack manifest`
    - `slack run`

#### Feedback

- Code implementation, what could be improved are there any technical issues with this approach
- I was able to manually test `slack run` for async apps but had issues implementing scenario tests for them, any ideas for this can help 
- User experience, are error clear and handled nicely, does this integrate well with the existing capabilities of the CLI
- Are there any higher level solutions I might have missed that are worth discussing
- Any concerns you may have on this, this addition has any moving components I very well may have missed something

### Category

* [ ] `slack_bolt.App` and/or its core components
* [ ] `slack_bolt.async_app.AsyncApp` and/or its core components
* [ ] Adapters in `slack_bolt.adapter`
* [ ] Document pages under `/docs`

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and have done my best effort to follow them.
